src/api/app/models/bs_request.rb

```diff
@@ -681,7 +681,7 @@ def change_review_state(new_review_state, opts = {})
     with_lock do
       new_review_state = new_review_state.to_sym
 
-      raise InvalidStateError, 'request is not in review state' unless state == :review || (state == :new && new_review_state == :new)
+      raise InvalidStateError, 'request is not in review state' unless state == :review || (state.in?([:new, :declined]) && new_review_state == :new)
```

Comment:
The exception message (`request is not in review state`) doesn't match the conditions in the `unless` statement. This should be adapted.